src/vm/assembly.cpp

```diff
@@ -2583,14 +2586,21 @@ HRESULT RunMain(MethodDesc *pFD ,
 
         if (pParam->pFD->IsVoid()) 
         {
-            // Set the return value to 0 instead of returning random junk
-            *pParam->piRetVal = 0;
             threadStart.Call(&stackVar);
+            *pParam->piRetVal =
+#ifndef FEATURE_CORECLR
+                // Set the return value to 0 instead of returning random junk
+                0;
```

Comment:
In desktop, the latched exit code can be changed by code running during shutdown after main exits (e.g. unload event handlers, code running on background threads). Not sure whether we want to preserve this behavior.